ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -1099,8 +1118,19 @@ func (r *RayServiceReconciler) reconcileServe(ctx context.Context, rayServiceIns
 			return ctrl.Result{RequeueAfter: ServiceDefaultRequeueDuration}, false, false, err
 		}
 
-		logger.Info("Mark cluster as unhealthy", "rayCluster", rayClusterInstance)
-		r.Recorder.Eventf(rayServiceInstance, "Normal", "Restarting", "The cluster will restart after %s", ServiceRestartRequeueDuration)
+		availableWorkerReplicas := rayClusterInstance.Status.AvailableWorkerReplicas
+		desiredWorkerReplicas := rayClusterInstance.Status.DesiredWorkerReplicas
+		logger.Info(
+			"Restart RayCluster", "AvailableWorkerReplicas", availableWorkerReplicas, "DesiredWorkerReplicas", desiredWorkerReplicas,
+			"restart reason",
+			"The serve application is unhealthy, restarting the cluster. If the AvailableWorkerReplicas is not equal to DesiredWorkerReplicas, "+
+				"this may imply that the Autoscaler does not have enough resources to scale up the cluster. Hence, the serve application does not "+
+				"have enough resources to run. Please check https://github.com/ray-project/kuberay/blob/master/docs/guidance/rayservice-troubleshooting.md for more details.",
+			"RayCluster", rayClusterInstance)
+		r.Recorder.Eventf(
+			rayServiceInstance, "Normal", "Restarting",
+			"Please check https://github.com/ray-project/kuberay/blob/master/docs/guidance/rayservice-troubleshooting.md for more details. The cluster will restart after %s", ServiceRestartRequeueDuration,
+			"AvailableWorkerReplicas", availableWorkerReplicas, "DesiredWorkerReplicas", desiredWorkerReplicas)
```

Comment:
Nit: we can check in the code whether available=desired and only print the last part if necessary, right? Could be less spammy that way

Reply:
Sure, I don't have a strong opinion.